app/models/host.rb

```diff
@@ -1939,17 +1939,15 @@ def self.node_types
   end
 
   def self.openstack_hosts_exists?
-    ems = ManageIQ::Providers::Openstack::InfraManager.pluck(:id)
-    ems.empty? ? false : Host.where(:ems_id => ems).exists?
+    ManageIQ::Providers::Openstack::InfraManager::Host.count > 0
   end
 
   def self.non_openstack_hosts_exists?
-    ems = ManageIQ::Providers::Openstack::InfraManager.pluck(:id)
-    Host.where.not(:ems_id => ems).exists?
+    (Host.count - ManageIQ::Providers::Openstack::InfraManager::Host.count) > 0
   end
 
   def openstack_host?
-    ext_management_system.class == ManageIQ::Providers::Openstack::InfraManager
+    kind_of?(ManageIQ::Providers::Openstack::InfraManager::Host)
```

Comment:
@Fryguy  This refactoring should not break anything, from what I see. Not sure why it is here, but I remember it was added for UI usage. :-)